Automattic/wp-calypso
# Site Editor Tracking - Template Part creation / replacement flows.

#### Changes proposed in this Pull Request

Adds tracking for various template part flows:

* Creating a new template part via the placeholder.  `wpcom_block_editor_create_template_part` with properties for `variation_slug` and `content` if applicable.  If the user has selected a pattern in the creation flow, `content` will correspond to that patterns markup.  Otherwise, the `content` property will be absent as the user selected to create a blank template part.
* Choosing an existing template part via the placeholder.  `wpcom_block_editor_template_part_choose_existing` with properties of `variation_slug` and `template_part_id` (such as "pub/tt1-blocks//header") of the selected template part.
* Replacing a template part via the block inserter.  `wpcom_block_editor_template_part_replace` will have 2 properties corresponding to the replaced item (`replaced_template_part_id` and `replaced_variation_slug`) and 2 properties corresponding to the newly selected item (`template_part_id` and `variation_slug`).

#### Testing instructions



* Follow steps to enable tracking debugging at PCYsg-nrf-p2
* Open the site editor.
* Add a template part block ( "Header" or "Footer")
* Using the placeholder, select "New header" (or footer).  
* Select "choose" from the pattern selector tool.  Verify the `wpcom_block_editor_create_template_part` event fires with the expected properties as noted above.
* Repeat the above but instead of selecting to "choose" a pattern, select "Start blank".  Verify the above event fires and there is no `content` property.
* Add another template part block.  This time, instead of selecting "New template part", select "choose existing" and select an existing template part from the dropdown.  Verify that `wpcom_block_editor_template_part_choose_existing` event fires with the expected properties listed above.
* Next, select an existing and instantiated template part in the editor (this could be one you created in a previous step).  With the parent template part block selected, select "replace" in the block toolbar:
![Screen Shot 2021-06-28 at 2 12 08 PM](https://user-images.githubusercontent.com/28742426/123690161-0f2c9600-d822-11eb-9faf-0eec09cc021b.png)
* Select a template part to replace with (note, another template part of this area/variation must exist).  Verify the `wpcom_block_editor_template_part_replace` event fires with the expected properties `replaced_template_part_id`, `replaced_variation_slug`, `template_part_id`, and `variation_slug`.



Related to https://github.com/Automattic/wp-calypso/issues/53410
